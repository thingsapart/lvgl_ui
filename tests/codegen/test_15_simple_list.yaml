# UI Test: A list with text headers and clickable icon buttons.
- type: list
  size: [180, 220]
  center: []
  children:
    - init: { lv_list_add_text: [File] }
    - init: { lv_list_add_button: [LV_SYMBOL_FILE, New] }
    - init: { lv_list_add_button: [LV_SYMBOL_SAVE, Save] }
    - init: { lv_list_add_text: [Actions] }
    - init: { lv_list_add_button: [LV_SYMBOL_CLOSE, Close] }
