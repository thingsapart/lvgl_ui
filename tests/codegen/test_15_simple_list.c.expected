/* AUTO-GENERATED by the 'c_code' backend */

#include "lvgl.h"
#include "c_gen/lvgl_dispatch.h" // For obj_registry_add
#include "data_binding.h"

void create_ui(lv_obj_t* parent) {
    // unnamed: list_0 (list)
    lv_obj_t* list_0 = lv_list_create(parent);

    lv_obj_set_style_size(list_0, 180, 220, 0);
    lv_obj_center(list_0);
    // unnamed: obj_1 (obj)
    lv_obj_t* obj_1 = lv_list_add_text(list_0, "File");

    // unnamed: obj_2 (obj)
    lv_obj_t* obj_2 = lv_list_add_button(list_0, "\xef\x85\x9b", "New");

    // unnamed: obj_3 (obj)
    lv_obj_t* obj_3 = lv_list_add_button(list_0, "\xef\x83\x87", "Save");

    // unnamed: obj_4 (obj)
    lv_obj_t* obj_4 = lv_list_add_text(list_0, "Actions");

    // unnamed: obj_5 (obj)
    lv_obj_t* obj_5 = lv_list_add_button(list_0, "\xef\x80\x8d", "Close");


}
